c++/src/kj/test-helpers.c++

```diff
@@ -153,5 +153,65 @@ bool expectFatalThrow(kj::Maybe<Exception::Type> type, kj::Maybe<StringPtr> mess
 #endif
 }
 
+bool expectExit(Maybe<int> statusCode, Function<void()> code) {
+#if _WIN32
+  // We don't support death tests on Windows due to lack of efficient fork.
+  return true;
+#else
+  pid_t child;
+  KJ_SYSCALL(child = fork());
+  if (child == 0) {
+    code();
+    _exit(0);
+  }
+
+  int status;
+  KJ_SYSCALL(waitpid(child, &status, 0));
+
+  if (WIFEXITED(status)) {
+    KJ_IF_MAYBE(s, statusCode) {
+      KJ_EXPECT(WEXITSTATUS(status) == *s);
+      return WEXITSTATUS(status) == *s;
+    } else {
+      KJ_EXPECT(WEXITSTATUS(status) != 0);
+      return WEXITSTATUS(status) != 0;
+    }
+  } else {
+    KJ_FAIL_EXPECT("subprocess didn't exit", status);
```

Comment:
The `status` value produced by `waitpid()` is pretty hard for a human to interpret, since it's a bitfield of a bunch of stuff. Could we check `WIFSIGNALED()` here and if so produce an error like "code raised a signal instead of exiting; signal = 11 (Segmentation Fault)"? (You can use `strsignal()` from the C library to convert the signal number to a name.)

Similarly below if we are expecting a signal and instead the code exits, we should say what the exit code was.